File: SiriusQuality/SiriusQuality-Priestly-Taylor/src/cpp/SiriusQuality_Priestly_Taylor/PotentialET_PTComponent.cpp
#include "PotentialET_PTComponent.h"
using namespace SiriusQuality-Priestly-Taylor;
PotentialET_PTComponent::PotentialET_PTComponent()
{
       
}


double PotentialET_PTComponent::getlambdaV(){ return this->lambdaV; }
double PotentialET_PTComponent::getpsychrometricConstant(){ return this->psychrometricConstant; }
double PotentialET_PTComponent::getAlpha(){ return this->Alpha; }
int PotentialET_PTComponent::getih(){ return this->ih; }

void PotentialET_PTComponent::setlambdaV(double _lambdaV)
{
    _PriestlyTaylor.setlambdaV(_lambdaV);
}
void PotentialET_PTComponent::setpsychrometricConstant(double _psychrometricConstant)
{
    _PriestlyTaylor.setpsychrometricConstant(_psychrometricConstant);
}
void PotentialET_PTComponent::setAlpha(double _Alpha)
{
    _PriestlyTaylor.setAlpha(_Alpha);
}
void PotentialET_PTComponent::setih(int _ih)
{
    _PriestlyTaylor.setih(_ih);
}
void PotentialET_PTComponent::Calculate_Model(PotentialET_PTState &s, PotentialET_PTState &s1, PotentialET_PTRate &r, PotentialET_PTAuxiliary &a, PotentialET_PTExogenous &ex)
{
    _PriestlyTaylor.Calculate_Model(s, s1, r, a, ex);
}
PotentialET_PTComponent::PotentialET_PTComponent(PotentialET_PTComponent& toCopy)
{
    lambdaV = toCopy.getlambdaV();
    psychrometricConstant = toCopy.getpsychrometricConstant();
    Alpha = toCopy.getAlpha();
    ih = toCopy.getih();
}

File: SiriusQuality/SiriusQuality-Priestly-Taylor/src/java/SiriusQuality_Priestly_Taylor/PotentialET_PTComponent.java
public class PotentialET_PTComponent
{
    
    public PotentialET_PTComponent() { }

    PriestlyTaylor _PriestlyTaylor = new PriestlyTaylor();

    public double getlambdaV()
    { return _PriestlyTaylor.getlambdaV(); }
    public void setlambdaV(double _lambdaV){
    _PriestlyTaylor.setlambdaV(_lambdaV);
    }

    public double getpsychrometricConstant()
    { return _PriestlyTaylor.getpsychrometricConstant(); }
    public void setpsychrometricConstant(double _psychrometricConstant){
    _PriestlyTaylor.setpsychrometricConstant(_psychrometricConstant);
    }

    public double getAlpha()
    { return _PriestlyTaylor.getAlpha(); }
    public void setAlpha(double _Alpha){
    _PriestlyTaylor.setAlpha(_Alpha);
    }

    public Integer getih()
    { return _PriestlyTaylor.getih(); }
    public void setih(Integer _ih){
    _PriestlyTaylor.setih(_ih);
    }
    public void  Calculate_Model(PotentialET_PTState s, PotentialET_PTState s1, PotentialET_PTRate r, PotentialET_PTAuxiliary a, PotentialET_PTExogenous ex)
    {
        _PriestlyTaylor.Calculate_Model(s, s1, r, a, ex);
    }
    private double lambdaV;
    private double psychrometricConstant;
    private double Alpha;
    private Integer ih;
    public PotentialET_PTComponent(PotentialET_PTComponent toCopy) // copy constructor 
    {
        this.lambdaV = toCopy.getlambdaV();
        this.psychrometricConstant = toCopy.getpsychrometricConstant();
        this.Alpha = toCopy.getAlpha();
        this.ih = toCopy.getih();

    }
}